Spam2: Digest for Unmoderated posts with settings

keshavsethi opened the pull request:
Here **digest for unmoderated posts** is added where I have used tag **digest:daily:spam** for daily digest and **digest:weekly:spam** for weekly digest. There is also a test button in the profile section which is used to generate the daily digest. Settings for this are only visible to moderators and admin. _**This PR is similar to https://github.com/publiclab/plots2/pull/7987 with some additional changes like tags and settings.**_

Please refer to following Screenshots:
![Screenshot from 2020-06-22 18-50-06](https://user-images.githubusercontent.com/36025262/85294961-e65e1c00-b4bc-11ea-9dff-0e8abe21535f.png)
![Screenshot from 2020-06-22 18-49-34](https://user-images.githubusercontent.com/36025262/85294966-e8c07600-b4bc-11ea-8c4f-a73e385a433a.png)
![Screenshot from 2020-06-22 18-49-51](https://user-images.githubusercontent.com/36025262/85294956-e3632b80-b4bc-11ea-8372-33b1c489c91c.png)
Please review @jywarren @cesswairimu @pydevsg @ananyaarun @ebarry @emilyashley @VladimirMikulic @Uzay-G 
Thanks!

jywarren commented on app/views/users/settings.html.erb:
```diff
@@ -90,6 +90,35 @@
   <br />
   <br />
 
+  <% if current_user.can_moderate? %>
```
Here, could we create a (maybe `<h4>` bounded?) subsection for Moderation settings? That way it's a little easier to skim to the settings you want. Also, could we add `a` in front of each instance of this phrase, for both existing and new settings? It's a little better grammatically:

`weekly digest` => `a weekly digest` (and same for all others on this page?

Thank you @keshavsethi this is looking AMAZING!!! 🎉 🙌

keshavsethi commented:
@jywarren 
![Screenshot from 2020-06-24 04-42-54](https://user-images.githubusercontent.com/36025262/85476187-59f34c80-b5d5-11ea-90fa-146738234666.png)
I have added a subsection for this and added `a` in some lines. Please review. 
Thanks!!

jywarren commented:
This is super! I'll be sure to mention this for staff; once it's live on the site, perhaps a brief blog post with this last screenshot could help introduce this new feature to folks on PublicLab.org? It'd be great to hear what they think!

cesswairimu commented:
:tada: :tada:
